apps/src/applab/applab.js

```diff
@@ -1080,15 +1144,37 @@ Applab.onSavePropertiesButton = function(el, event) {
   el.style.width = document.getElementById('design-property-width').value;
   el.style.height = document.getElementById('design-property-height').value;
   $(el).text(document.getElementById('design-property-text').value);
-  Applab.levelHtml = document.getElementById('divApplab').innerHTML;
+  Applab.levelHtml = Applab.serializeToLevelHtml();
 };
 
 Applab.onDeletePropertiesButton = function(el, event) {
   el.parentNode.removeChild(el);
-  Applab.levelHtml = document.getElementById('divApplab').innerHTML;
+  Applab.levelHtml = Applab.serializeToLevelHtml();
   Applab.clearProperties();
 };
 
+Applab.serializeToLevelHtml = function () {
+  var s = new XMLSerializer();
+  var divApplab = document.getElementById('divApplab');
+  var clone = divApplab.cloneNode(true);
+  // Remove unwanted classes added by jQuery.draggable.
+  // TODO(dave): Traverse children recursively once we support containers.
+  $(clone).children().removeAttr('class');
+  return s.serializeToString(clone);
+};
+
+Applab.parseFromLevelHtml = function(rootEl, isDesignMode) {
+  if (Applab.levelHtml) {
```

Comment:
Could invert this logic and reduce nesting: i.e.
if (!Applab.levelHtml) {
  return;
}
// do stuff